pkg/repo/callrepo.go

```diff
@@ -45,45 +48,57 @@ type CallsRepository struct {
 func NewCallsRepository(cfg *CallsConfig) (*CallsRepository, error) {
 	r := &CallsRepository{
 		calls: make(map[string]core.Handler),
+		mu:    &sync.Mutex{},
 	}
 
+	r.mu.Lock()
+	defer r.mu.Unlock()
```

Comment:
You don't need lock here, registry is not yet shared, you have exclusive access in constructor